protocol/messenger_handler.go

```diff
@@ -1541,6 +1541,19 @@ func (m *Messenger) HandleDeleteMessage(state *ReceivedMessageState, deleteMessa
 				return ErrInvalidDeletePermission
 			}
 		}
+
+		if chat.ChatType == ChatTypePrivateGroupChat || chat.ChatType == ChatTypePublic {
```

Comment:
Public chats don't have admins so you can remove the `ChatTypePublic` condition.

Also, you can it an `else if` with the condition above, since it cannot be a community chat and a group chat.